### PR TITLE
Mandos DNS test upgrade & some new syntax

### PR DESCRIPTION
**Pull request by andrei-marinica**
- DNS tests updated with the latest version
- Mandos address syntax enhanced:
smart contract address format: `sc:my_contract`,
with shard id: `sc:my_contract#12`
`"username"` field

### CODE REVIEW

File: mandos-go/json/integrationTests/example.scen.json [hunk 1]
```diff
@@ -55,12 +55,14 @@
                             "ESDTRoleNFTBurn"
                         ]
                     },
+                    "username": "str:myusername.elrond",
                     "storage": {},
                     "code": ""
                 },
                 "address:smart_contract_address": {
```
Comment (camilbancioiu) [inline mandos-go/json/integrationTests/example.scen.json:62]: Maybe use `sc:` ?

File: test/dns/output/dns.abi.json [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+            "inputs": [],
+            "outputs": [
+                {
+                    "type": "bytes"
+                }
+            ]
+        }
+    ],
+    "types": {}
+}
```
Comment (camilbancioiu) [inline test/dns/output/dns.abi.json:187]: Newline at EOF.

File: mandos-go/expression/interpreter/functions.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+		address := createAddressFromPrefix([]byte(tokens[0]), numLeadingZeros, 32)
+		address[31] = shardId
+		return address[:], nil
+	default:
+		return []byte{}, fmt.Errorf("only one shard id separator allowed in address expression. Got: `%s`", input)
+	}
+}
+
+// SCAddressNumLeadingZeros is the number of zer obytes every smart contract address begins with
+const SCAddressNumLeadingZeros = 8
```
Comment (camilbancioiu) [inline mandos-go/expression/interpreter/functions.go:62]: Move at the top of the file.

**Review by camilbancioiu — CHANGES_REQUESTED** [review]

File: mandos-go/json/integrationTests/example.scen.json (hunk 1 above)
Comment (andrei-marinica) [reply mandos-go/json/integrationTests/example.scen.json:62]: `sc:` is for smart contract addresses, this is the username

File: test/dns/output/dns.abi.json (hunk 2 above)
Comment (andrei-marinica) [reply test/dns/output/dns.abi.json:187]: hmm, this is auto-generated so the fix is somewhere else
probably no need for ABI here

File: mandos-go/json/parse/parseAccountsCheck.go [hunk 4]
```diff
@@ -49,7 +50,7 @@ func (p *Parser) processCheckAccount(acctRaw oj.OJsonObject) (*mj.CheckAccount,
 					return nil, errors.New("invalid ESDT map")
 				}
 				for _, esdtKvp := range esdtMap.OrderedKV {
-					tokenNameStr, err := p.ValueInterpreter.InterpretString(esdtKvp.Key)
+					tokenNameStr, err := p.ExprInterpreter.InterpretString(esdtKvp.Key)
```
Comment (camilbancioiu) [inline mandos-go/json/parse/parseAccountsCheck.go:53]: Ui așa:
```suggestion
					tokenNameStr, err := p.ExprInterpreter.InterpretString(esdtKvp.Key) // ceva
```

**Review by camilbancioiu — DISMISSED** [review]

File: mandos-go/json/integrationTests/example.scen.json (hunk 1 above)
Comment (camilbancioiu) [reply mandos-go/json/integrationTests/example.scen.json:62]: I meant `address:smart_contract_address` could use `sc:`.

**Review by sasurobert — APPROVED** [review]

**Review by camilbancioiu — APPROVED** [review]

**Pull request merged**